lib/net.c

```diff
@@ -219,6 +233,120 @@ ipv4_ntuple_filter_add(struct gatekeeper_if *iface, rte_be32_t dst_ip_be,
 	return 0;
 }
 
+static int
+ipv4_flow_add(struct gatekeeper_if *iface, rte_be32_t dst_ip_be,
+	rte_be16_t src_port_be, rte_be16_t src_port_mask_be,
+	rte_be16_t dst_port_be, rte_be16_t dst_port_mask_be,
+	uint8_t proto, uint16_t queue_id)
+{
+	struct rte_flow_attr attr = { .ingress = 1, };
+	struct rte_flow_action_queue queue = { .index = queue_id };
+	struct rte_flow_action action[2] = {
+		{
+			.type = RTE_FLOW_ACTION_TYPE_QUEUE,
+			.conf = &queue,
+	       	},
+		{
+			.type = RTE_FLOW_ACTION_TYPE_END,
+		}
+	};
+	struct rte_flow_item pattern[3] = {
+		{
+			.type = RTE_FLOW_ITEM_TYPE_ETH,
+			.spec = {
+				.type = rte_cpu_to_be_16(RTE_ETHER_TYPE_IPV4),
+			},
+			.mask = {
+				.type = 0xFFFF,
+			},
+		},
+		{
+			.type = RTE_FLOW_ITEM_TYPE_IPV4,
+			.spec = {
+				.hdr = {
+					.dst_addr = dst_ip_be,
+					.next_proto_id = proto,
+				}
+			},
+			.mask = {
+				.hdr = {
+					.dst_addr = 0xFFFFFFFF,
+					.next_proto_id = 0xFF,
+				}
+			},
+		},
```

Comment:
`.spec` and `.mask` are pointers, not structs. So the code must be rewritten similarly another comment discusses the same issue for `pattern[0]`.